File: 2.2 Software Technologies/04.1 JAVA-SYNTAX - EXERCISE/18.Phonebook/src/Phonebook.java
import java.util.LinkedHashMap;
import java.util.Scanner;

public class Phonebook {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String inputLine = scanner.nextLine();

        LinkedHashMap<String, String> phoneBook = new LinkedHashMap<>();

        while (!inputLine.equals("END")) {

            String[] inputParams = inputLine.split("\\s+");
            String command = inputParams[0];

            switch (command) {
                case "A":
                    String name = inputParams[1];
                    String phoneNumber = inputParams[2];
                    phoneBook.put(name, phoneNumber);
                    break;
                case "S":
                    name = inputParams[1];
                    if (!phoneBook.containsKey(name)) {
                        System.out.printf("Contact %s does not exist.\n", name);
                    } else {
                        System.out.printf("%s -> %s\n", name, phoneBook.get(name));
                    }
                    break;
            }

            inputLine = scanner.nextLine();
        }
    }
}

File: 3.1.1 C# Advanced/02.1 EXERCISE-SETS AND DICTIONARIES/05.Phonebook/Phonebook.cs
using System;
using System.Collections.Generic;

namespace _05.Phonebook
{
    public class Phonebook
    {
        public static void Main()
        {
            var input = Console.ReadLine();
            var phonebook = new Dictionary<string, string>();

            while (!input.Equals("search"))
            {
                var inputParams = input.Split('-');
                var name = inputParams[0];
                var phoneNumber = inputParams[1];

                if (!phonebook.ContainsKey(name))
                {
                    phonebook[name] = "";
                }

                phonebook[name] = phoneNumber;

                input = Console.ReadLine();
            }

            var nameToSearch = Console.ReadLine();

            while (!nameToSearch.Equals("stop"))
            {
                if (phonebook.ContainsKey(nameToSearch))
                {
                    Console.WriteLine($"{nameToSearch} -> {phonebook[nameToSearch]}");
                }
                else
                {
                    Console.WriteLine($"Contact {nameToSearch} does not exist.");
                }

                nameToSearch = Console.ReadLine();
            }
        }
    }
}
